src/registrar/views/utility/permission_views.py

```diff
@@ -27,6 +31,77 @@ class DomainPermissionView(DomainPermission, DetailView, abc.ABC):
     # variable name in template context for the model object
     context_object_name = "domain"
 
+    # Adds context information for user permissions
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        user = self.request.user
+        context["is_analyst_or_superuser"] = user.is_staff or user.is_superuser
+        # Stored in a variable for the linter
+        action = "analyst_action"
+        action_location = "analyst_action_location"
+        # Flag to see if an analyst is attempting to make edits
+        if action in self.request.session:
+            context[action] = self.request.session[action]
+        if action_location in self.request.session:
+            context[action_location] = self.request.session[action_location]
+
+        return context
+
+    def log_analyst_form_actions(
```

Comment:
I feel like this does not belong in DomainPermissionView, but rather in some DomainLoggingClass. Furthermore, we're using Audit Log which looks like it's already logging this info.

I would gut this out and move it into another PR, and park it for further investigation.